docusaurus/docs/React/guides/sdk-state-management.mdx

```diff
@@ -0,0 +1,146 @@
+---
+id: sdk-state-management
+title: SDK State Management
+---
+
+Most of the application state which is the main driver behind the UI changes of our chat SDK lives within our LLC ([`StreamChat`](https://github.com/GetStream/stream-chat-js)) - more specifically client and channel instances and is _NOT REACTIVE_ - our React SDK makes these state stores reactive by listening to the same events client does and copies it with the help of `useState` API after it has been changed.
+
+This is how the SDK state pipeline looks like behind the scenes:
+
+```txt
+client attaches WS event listeners -> React SDK attaches WS event listeners -> WS event received -> client/channel instance updates its own state -> React SDK pulls the updated state and updates it locally
+```
```

Comment:
Can we make a nice chart here? I can help with the visual part if you need me to!